src/app/directives/datatable/pagination.directive.spec.ts

```diff
@@ -0,0 +1,355 @@
+      setup({ filters: filters$, getModels: getModels() });
+      setPage(3);
+      assertPage(3);
+      filters$.next({ filter: { id: { eq: 1 } } });
+      assertPage(1);
+    });
+
+    it("should reset to 0 on sort change", () => {
+      generateModels({ numModels: 25, totalModels: 100 });
+      setup({ filters: {}, getModels: getModels() });
+      setPage(2);
+      assertPage(2);
+      sortColumn(0);
+      assertPage(1);
+    });
+  });
+
+  describe("sorting", () => {
+    function assertSort(sortKey: string, direction: Direction = "asc") {
+      expect(spec.directive["pageAndSort$"].getValue().sort).toEqual({
+        direction,
+        orderBy: sortKey as keyof MockModel,
+      });
+    }
+
+    it("should read sort key from datatable column template", () => {
+      generateModels();
+      spec = createDirective(
+        `
+        <ngx-datatable
+          bawDatatableDefaults
+          [bawDatatablePagination]="{ filters: filters, getModels: getModels }"
+        >
+          <ngx-datatable-column
+            prop="id"
+            sortKey="sortKey"
+          ></ngx-datatable-column>
+        </ngx-datatable>
+      `,
+        { hostProps: { filters: {}, getModels: getModels() } }
+      );
+      sortColumn(0);
+      assertSort("sortKey");
+    });
+
+    it("should use column prop from datatable column template", () => {
+      generateModels();
+      spec = createDirective(
+        `
+        <ngx-datatable
+          bawDatatableDefaults
+          [bawDatatablePagination]="{ filters: filters, getModels: getModels }"
+        >
+          <ngx-datatable-column prop="id"></ngx-datatable-column>
+        </ngx-datatable>
+      `,
+        { hostProps: { filters: {}, getModels: getModels() } }
+      );
+      sortColumn(0);
+      assertSort("id");
+    });
+
+    it("should use column prop from datatable column input", () => {
+      generateModels();
+      spec = createDirective(
+        `
+        <ngx-datatable
+          bawDatatableDefaults
+          [bawDatatablePagination]="{ filters: filters, getModels: getModels }"
+          [columns]="columns"
+        ></ngx-datatable>
+      `,
+        {
+          hostProps: {
+            filters: {},
+            getModels: getModels(),
+            columns: [{ prop: "id" }],
+          },
+        }
+      );
+      sortColumn(0);
+      assertSort("id");
+    });
+
+    it("should use column name from datatable column input", () => {
+      generateModels();
+      spec = createDirective(
+        `
+        <ngx-datatable
+          bawDatatableDefaults
+          [bawDatatablePagination]="{ filters: filters, getModels: getModels }"
+          [columns]="columns"
+        ></ngx-datatable>
+      `,
+        {
+          hostProps: {
+            filters: {},
+            getModels: getModels(),
+            columns: [{ name: "Id" }],
+          },
+        }
+      );
+      sortColumn(0);
+      assertSort("id");
```

Comment:
This is really testing datatables' conversion of name into prop.

I'd say column input should also support `sortKey` attribute as well. Add that test. From the code it looks like it would just work.